.github/workflows/ci_cd.yml

```diff
@@ -246,37 +246,25 @@ jobs:
           doc-artifact-name: "documentation-html"
 
   release:
-    if: github.event_name == 'push' && contains(github.ref, 'refs/tags')
-    needs: build
+    name: Release project
+    if: ${{ github.event_name == 'push' && contains(github.ref, 'refs/tags') }}
+    needs: [build]
     runs-on: ubuntu-latest
+    environment: release
+    permissions:
+      id-token: write
+      contents: write
     steps:
-      - name: Set up Python
-        uses: actions/setup-python@v5
+      - name: Release to the public PyPI repository
+        uses: ansys/actions/release-pypi-public@v6
         with:
-          python-version: ${{ env.MAIN_PYTHON_VERSION }}
-
-      - uses: actions/checkout@v4
-      - uses: actions/download-artifact@v4
-      
-      - name: List directory structure
-        run: ls -R
-
-      - name: Publish to PyPI
-        env:
-          TWINE_USERNAME: __token__
-          TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
-        run: |
-          python -m pip install --upgrade pip twine
-          python -m twine upload --skip-existing ./**/*.whl
-          python -m twine upload --skip-existing ./**/*.tar.gz
-      
+          library-name: ${{ env.PACKAGE_NAME }}
```

Comment:
You might have to define the package name :) I don't see it within the env vars available

Reply:
Ah right

Reply:
Also, I'm not sure you are using the ansys/build-library action. Instead you are building it yourself... so you might have to adapt the name of the artifact you are using

````yml
      - name: Upload wheel
        uses: actions/upload-artifact@v4
        with:
          path: dist/
          retention-days: 7
 ```
 
 The ``ansys/actions/release-pypi-public@v6`` action is expecting an artifact called ``${{ env.PACKAGE_NAME }}-artifacts``... so:
 
 ````yml
      - name: Upload wheel
        uses: actions/upload-artifact@v4
        with:
          name: ${{ env.PACKAGE_NAME }}-artifacts
          path: dist/
          retention-days: 7
 ```